h/views/oidc.py

```diff
@@ -70,20 +73,24 @@ def _decode_oauth2_state_param(state_param, key):
     return jwt.decode(state_param, key, algorithms=["HS256"])
 
 
-@view_defaults(request_method="GET", route_name="oidc.connect.orcid")
+@view_defaults(request_method="GET")
 class ORCIDConnectViews:
     def __init__(self, request: Request) -> None:
         self._request = request
 
-    @view_config(is_authenticated=True)
+    @view_config(is_authenticated=True, route_name="oidc.connect.orcid")
+    @view_config(is_authenticated=False, route_name="oidc.login.orcid")
```

Comment:
This single view now handles both the connect route (which is only available when logged in) and the login route (which is only available when _not_ logged in).